### PR TITLE
CBR-3095: Add User Permissions to Role

### PR DESCRIPTION
**Pull request by surabhi70**
Purpose: Add User permissions to sgas_sc_dev_admin

Note: Testing performed in DEV

### CODE REVIEW

File: src/json/federate-role_info.json [hunk 1]
```diff
@@ -114,4 +114,20 @@
     		  }		
   		]
 	}
+    "iamPermissionsPolicyDocument":{
```
Comment (jophine-antony) [inline src/json/federate-role_info.json]: This can be createUserIAMPolicyDocument

File: src/lib/awsIamRole.js [hunk 2]
```diff
@@ -38,25 +39,33 @@ var deleteIAMRole = function(iam, options, cb) {
...
+       if (options.roleName == process.env.SC_ADMIN_ROLE_NAME){
+         options.policyDocument = iamPermissionsPolicyDocument
+         addInlineRolePolicy(iam, options, cb);
+         attachRolePolicy(iam, options, cb);
+       }
+       if (typeof options.policyDocument === 'undefined' || options.policyDocument === null) {
+         attachRolePolicy(iam, options, cb);
+       }
+       else{
+         addInlineRolePolicy(iam, options, cb);
```
Comment (jophine-antony) [inline src/lib/awsIamRole.js:62]: why are we calling addInlineRolePolicy function twice for SC Admin role?

**Review by bhargavapratik — CHANGES_REQUESTED** [review]
Can you setup call to describe issue?
As I understand Code, we donot need to change LIB file to create / attache policy , we can do it in lamda function like we do for other IAM roles.

File: src/json/federate-role_info.json (hunk 1 above)
Comment (bhargavapratik) [inline src/json/federate-role_info.json]: Syntax error. correct it.

**Comment by surabhi70** [comment]
> Can you setup call to describe issue?
> As I understand Code, we donot need to change LIB file to create / attache policy , we can do it in lamda function like we do for other IAM roles.

When an account is created we drop a role "sgas_sc_admin" to it and attach a policy called as "AWSServiceCatalogAdminFullAccess" to that role. As a part of this change, we are just attaching one inline policy as well along with above policy.

File: src/lib/awsIamRole.js [hunk 3]
```diff
@@ -38,25 +39,33 @@ var deleteIAMRole = function(iam, options, cb) {
...
-      }
-    }
+   };
+   iam.createRole(params, function(err, data) {
+     if (err) {
+       return cb("Failed to create a role : " + err);
+     }
+     else {
+       options.roleArn = data.Role.Arn;
+       if (options.roleName == process.env.SC_ADMIN_ROLE_NAME){
```
Comment (SonalOjha) [inline src/lib/awsIamRole.js]: Kindly cook the data first and let this createRole take appropriate params to create the AWS IAM role. This function is a generic one having SC_ADMIN_ROLE_NAME check is not the correct way.

**Review by SonalOjha — CHANGES_REQUESTED** [review]

File: src/lib/awsIamRole.js [hunk 4]
```diff
@@ -38,25 +39,33 @@ var deleteIAMRole = function(iam, options, cb) {
 }
 */
 var createRole = function(iam, options, cb) {
-  var extid = options.externalId;
-  options.assumeRolePolicyDocument.Statement[0].Condition.StringEquals = {"sts:ExternalId": extid};
-  var params = {
+   var extid = options.externalId;
+   const awsroles = JSON.parse(fs.readFileSync('./json/federate-role_info.json', 'utf8'));
```
Comment (RajkumarShivage) [inline src/lib/awsIamRole.js]: Do we need this? We have existing functionality to create IAM roles and attach inline policy stored in federate-role_info.json.

**Review by bhargavapratik — APPROVED** [review]

File: src/lib/awsIamRole.js (hunk 4 above)
Comment (surabhi70) [reply src/lib/awsIamRole.js]: Please review the latest changes.

File: src/lib/awsIamRole.js (hunk 3 above)
Comment (surabhi70) [reply src/lib/awsIamRole.js]: Please review the latest changes.

File: src/json/federate-role_info.json [hunk 5]
```diff
@@ -113,5 +113,21 @@
       			"Resource": "arn:aws:ec2:*:*:instance/*"
     		  }		
   		]
-	}
+	},
+    "createUserIAMPolicyDocument":{
+        "Version": "2012-10-17",
+        "Statement": [
+        {
+         "Sid": "",
```
Comment (SonalOjha) [inline src/json/federate-role_info.json]: SID should not be blank it should be unique string, can add SCUserCreationPolicy

**Review by SonalOjha — APPROVED** [review]
Just a minor comment, but approving that it would get addressed.

File: src/json/federate-role_info.json (hunk 5 above)
Comment (surabhi70) [reply src/json/federate-role_info.json]: Done

**Review by RajkumarShivage — APPROVED** [review]
LGTM

**Pull request merged**